src/act.offensive.cpp

```diff
@@ -2474,3 +2474,174 @@ ACMD(do_stomp) {
+    ObjData *weapon;
+
+    if (ROOM_EFF_FLAGGED(ch->in_room, ROOM_EFF_DARKNESS) && !CAN_SEE_IN_DARK(ch)) {
+        char_printf(ch, "It is too dark!&0\n");
+        return;
+    }
+
+    if (!GET_SKILL(ch, SKILL_CARTWHEEL)) {
+        char_printf(ch, "You'd better leave all the tumbling to the rogues.\n");
+        return;
+    }
+
+    if (EFF_FLAGGED(ch, EFF_IMMOBILIZED)) {
+        char_printf(ch, "You don't have the mobility!\n");
+        return;
+    }
+
+    one_argument(argument, arg);
+
+    if (!(vict = find_char_in_room(&world[ch->in_room], find_vis_by_name(ch, arg)))) {
+        if (FIGHTING(ch))
+            vict = FIGHTING(ch);
+        else {
+            char_printf(ch, "&0Cartwheel at whom?&0\n");
+            return;
+        }
+    }
+
+    if (vict == ch) {
+        char_printf(ch, "You can't cartwheel into yourself!\n");
+        return;
+    }
+
+    /* check pk/pets/shapeshifts */
+    if (!attack_ok(ch, vict, true))
+        return;
+
+    if (CONFUSED(ch))
+        vict = random_attack_target(ch, vict, true);
+
+    if (MOB_FLAGGED(vict, MOB_NOBASH)) {
+        act("You &3&bslam&0 into $N, but $E seems quite unmoved.", false, ch, 0, vict, TO_CHAR);
+        act("$n &3&bslams&0 into $N, who seems as solid as a rock!", false, ch, 0, vict, TO_NOTVICT);
+        act("$n &3&bslams&0 into you, attempting to knock you down.", false, ch, 0, vict, TO_VICT);
+        /* A pause... but you don't fall down. */
+        WAIT_STATE(ch, PULSE_VIOLENCE * 2);
+        set_fighting(vict, ch, false);
+        return;
+    }
+
+    if (GET_SIZE(vict) - GET_SIZE(ch) > 1) {
+        char_printf(ch, "&7&bYou fall over as you try to knock down someone so large!&0\n");
+        act("&7&b$n BOUNCES off $N, as $e tries to knock down $N's much larger size.&0", false, ch, 0, vict,
+            TO_NOTVICT);
+        act("&7&b$n BOUNCES off you as $e tries to knock down your much larger size.&0", false, ch, 0, vict,
+            TO_VICT);
+        percent = prob + 1;
+    } else if (GET_SIZE(ch) - GET_SIZE(vict) > 2) {
+        char_printf(ch, "&7&bYou fall over as you try to knock down someone with such small size.&0\n");
+        act("&7&b$n trips over $N, as $e tries to knock down $N's much smaller size.&0", false, ch, 0, vict,
+            TO_NOTVICT);
+        act("&7&b$n trips over you as $e tries to knock down your much smaller size.&0", false, ch, 0, vict,
+            TO_VICT);
+        percent = prob + 1;
+    }
+
+    prob = GET_SKILL(ch, SKILL_CARTWHEEL);
+    prob += (dex_app_skill[GET_DEX(ch)].traps / 2);
+    prob += int_app[GET_INT(ch)].bonus;
+    prob += GET_HITROLL(ch) - monk_weight_penalty(ch);
+    percent = random_number(1, 100);
+    percent += GET_SKILL(vict, SKILL_DODGE);
+
+    if (prob > percent) {
+        if (damage_evasion(vict, ch, 0, DAM_CRUSH) || MOB_FLAGGED(vict, MOB_ILLUSORY)) {
+            act(EVASIONCLR "You cartwheel right through $N" EVASIONCLR " and fall in a heap on the other side!", false, ch, 0,
```

Comment:
Why do you keep setting the same color?

Reply:
I dunno, I'm just copy-pasting from extant bash code.